src/network/networkd-link.c

```diff
@@ -1020,6 +1020,12 @@ static int link_enter_set_addresses(Link *link) {
                                 log_link_warning_errno(link, r, "Failed to set NTP server for DHCP server, ignoring: %m");
                 }
 
+                if (link->network->dhcp_server_emit_router) {
+                        r = sd_dhcp_server_set_emit_router(link->dhcp_server, true);
```

Comment:
Why not do away with the conditional and call `sd_dhcp_server_set_emit_router(link->dhcp_server, link->network->dhcp_server_emit_router)` always? You also need `log_link_warning_errno` like in other places in this function.